src/modules/0.0.5/aws_ec2/index.ts

```diff
@@ -78,14 +89,37 @@ export const AwsEc2Module: Module = new Module({
           }
           return out;
         },
-        updateOrReplace: (_a: Instance, _b: Instance) => 'replace',
+        updateOrReplace: (prev: Instance, next: Instance) => {
+          if (!AwsEc2Module.utils.instanceEqReplaceableFields(prev, next)) return 'replace';
+          return 'update'
+        },
         update: async (es: Instance[], ctx: Context) => {
+          const client = await ctx.getAwsClient() as AWS;
           const out = [];
           for (const e of es) {
             const cloudRecord = ctx?.memo?.cloud?.Instance?.[e.instanceId ?? ''];
-            const created = await AwsEc2Module.mappers.instance.cloud.create([e], ctx);
-            await AwsEc2Module.mappers.instance.cloud.delete([cloudRecord], ctx);
-            out.push(created);
+            const isUpdate = AwsEc2Module.mappers.instance.cloud.updateOrReplace(cloudRecord, e) === 'update';
+            if (isUpdate) {
+              const insId = e.instanceId as string;
+              if (!AwsEc2Module.utils.instanceEqTags(e, cloudRecord) && e.instanceId && e.tags) {
+                await client.updateTags(insId, e.tags)
+              }
+              if (!Object.is(e.state, cloudRecord.state) && e.instanceId) {
+                if (cloudRecord.state === State.STOPPED && e.state === State.RUNNING) {
+                  await client.startInstance(insId);
+                } else if (cloudRecord.state === State.RUNNING && e.state === State.HIBERNATED) {
+                  await client.stopInstance(insId, true);
+                } else if (cloudRecord.state === State.RUNNING && e.state === State.STOPPED) {
+                  await client.stopInstance(insId);
+                } else {
+                  throw new Error(`Invalid instance state transition. From CLOUD state ${cloudRecord.state} to DB state ${e.state}`);
```

Comment:
Not sure if I'd throw here, maybe just "restore" the cloud value and do nothing?

Reply:
this *should* never happen so would prefer to throw an error and if it comes up to often then address?